core/domain/prod_validation_jobs_one_off.py

```diff
@@ -1662,6 +1664,275 @@ def _get_custom_validation_functions(cls):
+                user_models.UserSettingsModel, [item.committer_id])
+        }
+
+
+class SkillSnapshotContentModelValidator(
+        BaseSnapshotContentModelValidator):
+    """Class for validating SkillSnapshotContentModel."""
+
+    related_model_name = 'skill'
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'skill_ids': (
+                skill_models.SkillModel,
+                [item.id[:item.id.find('-')]]),
+        }
+
+
+class SkillRightsModelValidator(BaseModelValidator):
+    """Class for validating SkillRightsModel."""
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        snapshot_model_ids = [
+            '%s-%d' % (item.id, version) for version in range(
+                1, item.version + 1)]
+        return {
+            'skill_ids': (
+                skill_models.SkillModel, [item.id]),
+            'creator_user_ids': (
+                user_models.UserSettingsModel, [item.creator_id]),
+            'snapshot_metadata_ids': (
+                skill_models.SkillRightsSnapshotMetadataModel,
+                snapshot_model_ids),
+            'snapshot_content_ids': (
+                skill_models.SkillRightsSnapshotContentModel,
+                snapshot_model_ids),
+        }
+
+
+class SkillRightsSnapshotMetadataModelValidator(
+        BaseSnapshotMetadataModelValidator):
+    """Class for validating SkillRightsSnapshotMetadataModel."""
+
+    related_model_name = 'skill rights'
+
+    @classmethod
+    def _get_change_domain_class(cls, unused_item):
+        return skill_domain.SkillRightsChange
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'skill_rights_ids': (
+                skill_models.SkillRightsModel,
+                [item.id[:item.id.find('-')]]),
+            'committer_ids': (
+                user_models.UserSettingsModel, [item.committer_id])
+        }
+
+
+class SkillRightsSnapshotContentModelValidator(
+        BaseSnapshotContentModelValidator):
+    """Class for validating SkillRightsSnapshotContentModel."""
+
+    related_model_name = 'skill rights'
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'skill_rights_ids': (
+                skill_models.SkillRightsModel,
+                [item.id[:item.id.find('-')]]),
+        }
+
+
+class SkillCommitLogEntryModelValidator(BaseCommitLogEntryModelValidator):
+    """Class for validating SkillCommitLogEntryModel."""
+
+    related_model_name = 'skill'
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        # Valid id: [skill/rights]-[skill_id]-[skill_version].
+        regex_string = '^(skill|rights)-%s-\\d*$' % (
+            item.skill_id)
+
+        return regex_string
+
+    @classmethod
+    def _get_change_domain_class(cls, item):
+        if item.id.startswith('rights'):
+            return skill_domain.SkillRightsChange
+        else:
```

Comment:
I have added the if else check but if both the cases fail, I feel it is better to just return None instead of raising an Exception since the error for invalid id will be caught by the id regex check. What do you think?

Reply:
Sorry, I can't remember what we decided on during the meeting we had just now, but feel free to go with whatever the decision was!

Reply:
We decided to return None in the case of invalid id, add a comment with the return statement and update the commit commands check to append an error message providing complete command list and specifying that no commit command domain object is found. I have updated the jobs and tests according to that.